lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetMetadataUtils.java

```diff
@@ -27,18 +27,15 @@
 // Based on org.apache.parquet.format.converter.ParquetMetadataConverter
 public class ParquetMetadataUtils
 {
-    private static final long MAX_STATS_SIZE = 4096; // limit stats to 4k
-
     private ParquetMetadataUtils() {}
 
     public static <T extends Comparable<T>> Statistics toParquetStatistics(org.apache.parquet.column.statistics.Statistics<T> stats, int truncateLength)
     {
         Statistics formatStats = new Statistics();
-        // Don't write stats larger than the max size rather than truncating. The
-        // rationale is that some engines may use the minimum value in the page as
-        // the true minimum for aggregations and there is no way to mark that a
-        // value has been truncated and is a lower bound and not in the page.
-        if (!stats.isEmpty() && withinLimit(stats, truncateLength)) {
```

Comment:
I don't fully understand. The parquet-mr code has usable truncation logic, just it also has this condition.
would it be sufficient to use parquet-mr code without coping, just provide truncateLength smaller than MAX_STATS_SIZE?

Reply:
We could do that, however it relies on the knowledge that MAX_STATS_SIZE in parquet is bigger than the truncation length that we want to apply. It could break if parquet-mr decreases MAX_STATS_SIZE or we increase truncation length. This could've been avoided if parquet-mr hadn't hard-coded MAX_STATS_SIZE.
Given the current code, it's clearer if we take control of this logic on our side and remove `withinLimit` altogether.

Reply:
we bundle parquet library so we don't need to consider all sort of possibilities that could happen
MAX_STATS_SIZE is public and can be inspected. and its value (4k) is enough for our needs.

> t's clearer if we take control of this logic on our side 

i agree it's clearer, but it comes with a high tax too -- we now have copy not only of the code we want to control, but also of the sibling methods & concepts (like sort orders)

i would rather avoid copying that code

Reply:
I've updated this to avoid copying now